### PR TITLE
add support for solutions and test files

### PR DESCRIPTION
**Pull request by caseycrogers**
As the title suggests, this allows users to specify content for dart pad's `solutions` and `tests` files.

I've also nuked HTML escaping and the file parser as all that code was specific to dartpad's code injector used by Dart Doc.

### CODE REVIEW

File: lib/dart_pad_widget.dart [hunk 1]
```diff
@@ -69,6 +60,14 @@ class DartPad extends StatefulWidget {
   String get iframeStyle {
     return "width:${width}px;height:${height}px;";
   }
+
+  Map<String, String> get sourceCodeFileMap {
+    return {
+      'main.dart': code,
+      if (testCode != null) 'test.dart': testCode!,
```
Comment (TimWhiting) [inline lib/dart_pad_widget.dart:123]: Where is solutionCode used? I assume you just forgot to add it to this map?

**Comment by TimWhiting** [comment]
Can you also add some docs for these parameters with a simple sample of what should be in code vs testCode, etc. You can probably just take an example from the dart-pad embedding docs. Additionally adding a test / solution / hint to the first sample in the example would be nice

File: lib/dart_pad_widget.dart (hunk 1 above)
Comment (caseycrogers) [reply lib/dart_pad_widget.dart:123]: My bad, I couldn't seem to get solution code to do anything so I've just snipped it for now.

**Comment by caseycrogers** [comment]
> Can you also add some docs for these parameters with a simple sample of what should be in code vs testCode, etc. You can probably just take an example from the dart-pad embedding docs. Additionally adding a test / solution / hint to the first sample in the example would be nice

Working on this now.

**Comment by caseycrogers** [comment]
I figure out how solution cod is supposed to work and brought it back in. Also added a ton of documentation and a new example for the new functionality.

**Comment by TimWhiting** [comment]
Merged, documentation generated and published as 2.1.0

**Pull request merged**